### PR TITLE
Handle Task commits (and Task/Project PR events) coming directly from GitHub.

### PR DESCRIPTION
**Pull request by jgerigmeyer**
## Cute animal pic
_Because everyone likes pictures of animals._

slow loris

## Link to Trello card (if applicable)
_If this PR relates to a Trello card, don't forget to reference this PR on the card as well._

https://trello.com/c/mykQfEsb

## Description
_The commit messages say what you did; this should explain why and how._
- [x] Description is in Trello card

## Steps to test/reproduce
_Please explain how to best reproduce the issue and/or test the changes locally._

1. Enable GH webhooks locally (see `CONTRIBUTING.md`)
2. Create a new Task.
3. Create an Org on that task to trigger branch-creation.
4. Go to the task branch (on https://github.com/oddbird/Metecho-Test/) and make a new commit directly from GitHub.
5. See the commit appear in Metecho, along with the ability to Submit for Review.
6. Create a PR for that branch, again directly from GitHub.
7. See the "Submit for Review" btn disappear, and the "View Pull Request" button appear.

## Impacted areas of website
_Provide URLs and/or a description of where to look._

- Task detail page

---

## Reviewer tasks:

- [ ] Pulled branch and ran code locally (or viewed on preview deployment, if applicable)
- [ ] Reproduced the issue to review in the browser
- [ ] Reviewed code
- [ ] Updated Trello card (if applicable)

### CODE REVIEW

**Comment by jgerigmeyer** [comment]
@wlonk I'm sure this could use some cleanup (and definitely some tests), but I'd love your review on it first. ...And then your code cleanup and tests, if applicable! 😄 

The basic goal is to allow Metecho to adjust `has_unmerged_commits` and `pr_number` accordingly when events come in via the GH webhook.

**Comment by jgerigmeyer** [comment]
[Chore: Confirm that commits work even if coming directly from GH (not via ME).](https://trello.com/c/mykQfEsb/179-chore-confirm-that-commits-work-even-if-coming-directly-from-gh-not-via-me)

File: metecho/api/hook_serializers.py [hunk 1]
```diff
@@ -37,22 +45,36 @@ class PrHookSerializer(HookSerializerMixin, serializers.Serializer):
...
-
     def _get_matching_instance(self, repository):
+        pr_number = self.validated_data["number"]
+        pr_head_ref = self.validated_data["pull_request"]["head"]["ref"]
+        pr_base_ref = self.validated_data["pull_request"]["base"]["ref"]
         return (
             Task.objects.filter(
-                project__repository=repository, pr_number=self.validated_data["number"]
+                project__repository=repository, pr_number=pr_number
+            ).first()
```
Comment (wlonk) [inline metecho/api/hook_serializers.py:65]: Prefer `.exists()` for semantic reasons.

Comment (wlonk) [reply metecho/api/hook_serializers.py:65]: Wait that was dumb, we need the instance, not whether it exists! This is good!

**Comment by wlonk** [comment]
Tests will be incoming.

**Pull request merged**